### PR TITLE
[extractor/artetv] Remove duplicate stream urls

### LINKED ISSUE #4510: [ArteTV]: Duplicate formats, incorrect "More Info" column
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.07.18** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

France, Germany, ...

### Provide a description that is worded well enough to be understood

In the formats listed with "--list-formats/-F" on ArteTV several video-only formats and sometimes some audio-only formats seem to be duplicates. md5sum/sha256sum confirms that different video-only files for supposedly different formats have the same contents.

In the example below from https://www.arte.tv/en/videos/108210-043-A/mapping-the-world-essentials/ all six VOF-*409 video formats have the same content. Same for all other video resolutions. Also all six audio-only formats in this example have the same content (French).

So next to the duplicates, also the "More Info" column contents are incorrect.

Tested on recent master (871a8929bcc3).

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ yt-dlp -vU -F "https://www.arte.tv/en/videos/108210-043-A/mapping-the-world-essentials/"
[debug] Command-line config: ['-vU', '-F', 'https://www.arte.tv/en/videos/108210-043-A/mapping-the-world-essentials/']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.07.18 [135f05ef6] (zip)
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python 3.10.5 (CPython 64bit) - Linux-5.18.0-2-amd64-x86_64-with-glibc2.33 (glibc 2.33)
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1-3 (setts), ffprobe 5.0.1-3, phantomjs 2.1.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2020.06.20, mutagen-1.45.1, pyxattr-0.7.2, secretstorage-3.3.2, sqlite3-2.6.0, websockets-10.2
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.07.18, Current version: 2022.07.18
yt-dlp is up to date (2022.07.18)
[debug] [ArteTV] Extracting URL: https://www.arte.tv/en/videos/108210-043-A/mapping-the-world-essentials/
[ArteTV] 108210-043-A: Downloading JSON metadata
[ArteTV] 108210-043-A: Downloading m3u8 information
[ArteTV] 108210-043-A: Downloading m3u8 information
[ArteTV] 108210-043-A: Downloading m3u8 information
[ArteTV] 108210-043-A: Downloading m3u8 information
[ArteTV] 108210-043-A: Downloading m3u8 information
[ArteTV] 108210-043-A: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 108210-043-A:
ID                               EXT RESOLUTION FPS │  FILESIZE   TBR PROTO │ VCODEC        VBR ACODEC     MORE INFO
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
VOF-STMF-program_audio_0-VOF     mp4 audio only     │                 m3u8  │ audio only        unknown    [fr] Original french version - closed captioning (FR) [ccFR]
VOF-STE_ESP_-program_audio_0-VOF mp4 audio only     │                 m3u8  │ audio only        unknown    [fr] Spanish (Subtitles) [OGsub-ESP]
VOF-STE_ITA_-program_audio_0-VOF mp4 audio only     │                 m3u8  │ audio only        unknown    [fr] Italian (Subtitles) [OGsub-ITA]
VOF-STE_POL_-program_audio_0-VOF mp4 audio only     │                 m3u8  │ audio only        unknown    [fr] Polish (Subtitles) [OGsub-POL]
VOF-program_audio_0-VOF          mp4 audio only     │                 m3u8  │ audio only        unknown    [fr] French (Original) [FR]
VOF-STE_ANG_-program_audio_0-VOF mp4 audio only     │                 m3u8  │ audio only        unknown    [fr] English (Subtitles) [OGsub-ANG]
VOF-STMF-409                     mp4 384x216     25 │ ~ 7.50MiB  410k m3u8  │ avc1.42e00d  410k video only Original french version - closed captioning (FR) [ccFR]
VOF-STMF-791                     mp4 640x360     25 │ ~14.49MiB  791k m3u8  │ avc1.4d401e  791k video only Original french version - closed captioning (FR) [ccFR]
VOF-STMF-1244                    mp4 768x432     25 │ ~22.79MiB 1245k m3u8  │ avc1.4d401e 1245k video only Original french version - closed captioning (FR) [ccFR]
VOF-STMF-1962                    mp4 1280x720    25 │ ~35.93MiB 1962k m3u8  │ avc1.4d401f 1962k video only Original french version - closed captioning (FR) [ccFR]
VOF-STE_ESP_-409                 mp4 384x216     25 │ ~ 7.50MiB  410k m3u8  │ avc1.42e00d  410k video only Spanish (Subtitles) [OGsub-ESP]
VOF-STE_ITA_-409                 mp4 384x216     25 │ ~ 7.50MiB  410k m3u8  │ avc1.42e00d  410k video only Italian (Subtitles) [OGsub-ITA]
VOF-STE_POL_-409                 mp4 384x216     25 │ ~ 7.50MiB  410k m3u8  │ avc1.42e00d  410k video only Polish (Subtitles) [OGsub-POL]
VOF-STE_ESP_-791                 mp4 640x360     25 │ ~14.49MiB  791k m3u8  │ avc1.4d401e  791k video only Spanish (Subtitles) [OGsub-ESP]
VOF-STE_ITA_-791                 mp4 640x360     25 │ ~14.49MiB  791k m3u8  │ avc1.4d401e  791k video only Italian (Subtitles) [OGsub-ITA]
VOF-STE_POL_-791                 mp4 640x360     25 │ ~14.49MiB  791k m3u8  │ avc1.4d401e  791k video only Polish (Subtitles) [OGsub-POL]
VOF-STE_ESP_-1244                mp4 768x432     25 │ ~22.79MiB 1245k m3u8  │ avc1.4d401e 1245k video only Spanish (Subtitles) [OGsub-ESP]
VOF-STE_ITA_-1244                mp4 768x432     25 │ ~22.79MiB 1245k m3u8  │ avc1.4d401e 1245k video only Italian (Subtitles) [OGsub-ITA]
VOF-STE_POL_-1244                mp4 768x432     25 │ ~22.79MiB 1245k m3u8  │ avc1.4d401e 1245k video only Polish (Subtitles) [OGsub-POL]
VOF-STE_ESP_-1962                mp4 1280x720    25 │ ~35.93MiB 1962k m3u8  │ avc1.4d401f 1962k video only Spanish (Subtitles) [OGsub-ESP]
VOF-STE_ITA_-1962                mp4 1280x720    25 │ ~35.93MiB 1962k m3u8  │ avc1.4d401f 1962k video only Italian (Subtitles) [OGsub-ITA]
VOF-STE_POL_-1962                mp4 1280x720    25 │ ~35.93MiB 1962k m3u8  │ avc1.4d401f 1962k video only Polish (Subtitles) [OGsub-POL]
VOF-409                          mp4 384x216     25 │ ~ 7.50MiB  410k m3u8  │ avc1.42e00d  410k video only French (Original) [FR]
VOF-791                          mp4 640x360     25 │ ~14.49MiB  791k m3u8  │ avc1.4d401e  791k video only French (Original) [FR]
VOF-1244                         mp4 768x432     25 │ ~22.79MiB 1245k m3u8  │ avc1.4d401e 1245k video only French (Original) [FR]
VOF-1962                         mp4 1280x720    25 │ ~35.93MiB 1962k m3u8  │ avc1.4d401f 1962k video only French (Original) [FR]
VOF-STE_ANG_-409                 mp4 384x216     25 │ ~ 7.50MiB  410k m3u8  │ avc1.42e00d  410k video only English (Subtitles) [OGsub-ANG]
VOF-STE_ANG_-791                 mp4 640x360     25 │ ~14.49MiB  791k m3u8  │ avc1.4d401e  791k video only English (Subtitles) [OGsub-ANG]
VOF-STE_ANG_-1244                mp4 768x432     25 │ ~22.79MiB 1245k m3u8  │ avc1.4d401e 1245k video only English (Subtitles) [OGsub-ANG]
VOF-STE_ANG_-1962                mp4 1280x720    25 │ ~35.93MiB 1962k m3u8  │ avc1.4d401f 1962k video only English (Subtitles) [OGsub-ANG]
$
$ yt-dlp -vU -f all -o "%(title)s [%(id)s]-%(format)s.%(ext)s" "https://www.arte.tv/en/videos/108210-043-A/mapping-the-world-essentials/"
[...]
$ md5sum *.mp4 | sort
85166a8b1647eb4f2654a61b602805c2  Finland and Sweden Opt for NATO [108210-043-A]-VOF-791 - 640x360 (French (Original) [FR]).mp4
85166a8b1647eb4f2654a61b602805c2  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ANG_-791 - 640x360 (English (Subtitles) [OGsub-ANG]).mp4
85166a8b1647eb4f2654a61b602805c2  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ESP_-791 - 640x360 (Spanish (Subtitles) [OGsub-ESP]).mp4
85166a8b1647eb4f2654a61b602805c2  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ITA_-791 - 640x360 (Italian (Subtitles) [OGsub-ITA]).mp4
85166a8b1647eb4f2654a61b602805c2  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_POL_-791 - 640x360 (Polish (Subtitles) [OGsub-POL]).mp4
85166a8b1647eb4f2654a61b602805c2  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STMF-791 - 640x360 (Original french version - closed captioning (FR) [ccFR]).mp4
d3f2da1b1f84f71c2f6c0b5d19a7f3cd  Finland and Sweden Opt for NATO [108210-043-A]-VOF-1962 - 1280x720 (French (Original) [FR]).mp4
d3f2da1b1f84f71c2f6c0b5d19a7f3cd  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ANG_-1962 - 1280x720 (English (Subtitles) [OGsub-ANG]).mp4
d3f2da1b1f84f71c2f6c0b5d19a7f3cd  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ESP_-1962 - 1280x720 (Spanish (Subtitles) [OGsub-ESP]).mp4
d3f2da1b1f84f71c2f6c0b5d19a7f3cd  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ITA_-1962 - 1280x720 (Italian (Subtitles) [OGsub-ITA]).mp4
d3f2da1b1f84f71c2f6c0b5d19a7f3cd  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_POL_-1962 - 1280x720 (Polish (Subtitles) [OGsub-POL]).mp4
d3f2da1b1f84f71c2f6c0b5d19a7f3cd  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STMF-1962 - 1280x720 (Original french version - closed captioning (FR) [ccFR]).mp4
e0bd18cb418970022c5041ababb6dc4f  Finland and Sweden Opt for NATO [108210-043-A]-VOF-program_audio_0-VOF - audio only (French (Original) [FR]).mp4
e0bd18cb418970022c5041ababb6dc4f  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ANG_-program_audio_0-VOF - audio only (English (Subtitles) [OGsub-ANG]).mp4
e0bd18cb418970022c5041ababb6dc4f  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ESP_-program_audio_0-VOF - audio only (Spanish (Subtitles) [OGsub-ESP]).mp4
e0bd18cb418970022c5041ababb6dc4f  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ITA_-program_audio_0-VOF - audio only (Italian (Subtitles) [OGsub-ITA]).mp4
e0bd18cb418970022c5041ababb6dc4f  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_POL_-program_audio_0-VOF - audio only (Polish (Subtitles) [OGsub-POL]).mp4
e0bd18cb418970022c5041ababb6dc4f  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STMF-program_audio_0-VOF - audio only (Original french version - closed captioning (FR) [ccFR]).mp4
ea21b3c1c511d8c5e75ed2b85ed2dec6  Finland and Sweden Opt for NATO [108210-043-A]-VOF-409 - 384x216 (French (Original) [FR]).mp4
ea21b3c1c511d8c5e75ed2b85ed2dec6  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ANG_-409 - 384x216 (English (Subtitles) [OGsub-ANG]).mp4
ea21b3c1c511d8c5e75ed2b85ed2dec6  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ESP_-409 - 384x216 (Spanish (Subtitles) [OGsub-ESP]).mp4
ea21b3c1c511d8c5e75ed2b85ed2dec6  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ITA_-409 - 384x216 (Italian (Subtitles) [OGsub-ITA]).mp4
ea21b3c1c511d8c5e75ed2b85ed2dec6  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_POL_-409 - 384x216 (Polish (Subtitles) [OGsub-POL]).mp4
ea21b3c1c511d8c5e75ed2b85ed2dec6  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STMF-409 - 384x216 (Original french version - closed captioning (FR) [ccFR]).mp4
f14045181a869f6d436bf4f829d4d789  Finland and Sweden Opt for NATO [108210-043-A]-VOF-1244 - 768x432 (French (Original) [FR]).mp4
f14045181a869f6d436bf4f829d4d789  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ANG_-1244 - 768x432 (English (Subtitles) [OGsub-ANG]).mp4
f14045181a869f6d436bf4f829d4d789  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ESP_-1244 - 768x432 (Spanish (Subtitles) [OGsub-ESP]).mp4
f14045181a869f6d436bf4f829d4d789  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_ITA_-1244 - 768x432 (Italian (Subtitles) [OGsub-ITA]).mp4
f14045181a869f6d436bf4f829d4d789  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STE_POL_-1244 - 768x432 (Polish (Subtitles) [OGsub-POL]).mp4
f14045181a869f6d436bf4f829d4d789  Finland and Sweden Opt for NATO [108210-043-A]-VOF-STMF-1244 - 768x432 (Original french version - closed captioning (FR) [ccFR]).mp4
```

### PR DESCRIPTION
**Pull request by Grub4K**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Remove duplicate streams if it is a subtitle only stream (`cc` or `OGsub`).

Fixes #4510

<details open><summary>Template</summary>

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Comment by Lesmiscore** [comment]
`self._remove_duplicate_formats` removes the duplicates by URLs

**Comment by Grub4K** [comment]
The problem with removing urls blindly would be that we want to preserve the original (`French (Original) [FR]`) only since it provides the correct data.
This might not happen if we remove on all formats since its not always the first in the source data.
To circumvent this I removed manually based on if it is a subtitle instead.

We could simplify a bit using this though, provided formats that come first will always be preserved:
```
formats.extend(potential_duplicates)
self._remove_duplicate_formats(formats)
self._sort_formats(formats)
```

File: yt_dlp/extractor/arte.py [hunk 1]
```diff
@@ -135,6 +135,7 @@ def _real_extract(self, url):
                 'Video is not available in this language edition of Arte or broadcast rights expired', expected=True)
 
         formats, subtitles = [], {}
+        potential_duplicates = []
```
Comment (pukkandan) [inline yt_dlp/extractor/arte.py:138]: The var name isnt clear imo. potential duplicate what? `potential_duplicate_formats` works. Or maybe `secondary_formats`?

File: yt_dlp/extractor/arte.py [hunk 2]
```diff
@@ -152,22 +153,26 @@ def _real_extract(self, url):
                     not m.group('sdh_sub'),                 # and we prefer not the hard-of-hearing subtitles if there are subtitles
                 )))
 
+            short_label = traverse_obj(stream_version, "shortLabel", expected_type=str, default='?')
```
Comment (pukkandan) [inline yt_dlp/extractor/arte.py:156]:
```suggestion
            short_label = traverse_obj(stream_version, 'shortLabel', expected_type=str, default='?')
```

**Review by pukkandan — APPROVED** [review]
Otherwise lgtm

File: yt_dlp/extractor/arte.py (hunk 2 above)
Comment (Grub4K) [reply yt_dlp/extractor/arte.py:156]: Right, that was copied from the old source, which uses double quotes.

File: yt_dlp/extractor/arte.py (hunk 1 above)
Comment (Grub4K) [reply yt_dlp/extractor/arte.py:138]: `secondary_formats` seems better, that way we get `formats.extend(secondary_formats)`

**Pull request merged**